backend/app/graphql/all_users.py

```diff
@@ -0,0 +1,46 @@
+import graphene
+from ..graphql.services import services
+from .types import QueryList, Query
+
+
+class User(Query):
+    name = graphene.String()
+    email = graphene.String()
+    role = graphene.String()
+
+
+class UserQueries(QueryList):
+    users = graphene.List(
+        User,
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+        role=graphene.String(default_value=""),
+    )
+
+    def resolve_users(self, info, first, offset, role):
+        user_service = services["user_service"]
+        users = user_service.get_users()
+
+        if role != "":
+            filtered = []
+            for user in users:
```

Comment:
Instead of manually looping over all users, it would be better to use MongoDB's built-in filtering (likely to be more efficient since it's written in C++). For an example: https://github.com/uwblueprint/feeding-canadian-kids/pull/26/files#diff-728eb19f5e1cb2ece9f624da0891ea7ffb16f3ed0810bda4e6206086d4e20e4aR49